apps/emqx_authorization/src/emqx_authorization.erl

```diff
@@ -0,0 +1,227 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2020-2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_authorization).
+
+-include("emqx_authorization.hrl").
+
+-export([ register_metrics/0
+        , init/0
+        , compile/1
+        , lookup/0
+        , update/1
+        , check_authorization/5
+        , match/4
+        ]).
+
+-spec(register_metrics() -> ok).
+register_metrics() ->
+    lists:foreach(fun emqx_metrics:ensure/1, ?ACL_METRICS).
+
+init() ->
+    ok = register_metrics(),
+    % {ok, Conf} = hocon:load("etc/plugins/emqx_authorization.conf",#{format => richmap}),
+    RawConf = proplists:get_value(rules, application:get_all_env(?APP), []),
+    {ok, MapConf} = hocon:binary(jsx:encode(#{rules => RawConf}), #{format => richmap}),
```

Comment:
no need to encode and then load binary again if we call `hocon_schema:check_plain`